sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/shuffle/UnsafeRowReceiver.scala

```diff
@@ -41,11 +50,15 @@ private[shuffle] case class ReceiverEpochMarker() extends UnsafeRowReceiverMessa
  */
 private[shuffle] class UnsafeRowReceiver(
       queueSize: Int,
+      numShuffleWriters: Int,
+      checkpointIntervalMs: Long,
       override val rpcEnv: RpcEnv)
     extends ThreadSafeRpcEndpoint with ContinuousShuffleReader with Logging {
   // Note that this queue will be drained from the main task thread and populated in the RPC
   // response thread.
-  private val queue = new ArrayBlockingQueue[UnsafeRowReceiverMessage](queueSize)
+  private val queues = Array.fill(numShuffleWriters) {
```

Comment:
Queue per writer makes it easier to handle the alignment. However, if the queues do not get filled evenly, it makes it much more difficult to set a single size across all the queues. In a single queue this would not be much of an issue since the queues are shared by all writers.

Reply:
Well, is it a problem if we don't set a single size across all the queues?

Reply:
Then it would be configuration nightmare..It should be ok to start with the proposed approach and use same size for all the queues. We can change it later if we see issues as long as the internal details are not exposed to the user.

Reply:
I agree. This level of issues can be dealt with later once we run this and observed the issues for real. Also, it's likely that we will eventually replace a whole lot of this with a non-RPC-endpoint-based transfer mechanism. So lets keep something very simple for now.

Reply:
Hi TD, just a question here, what's the 'a non-RPC-endpoint-based transfer mechanism' refers for?